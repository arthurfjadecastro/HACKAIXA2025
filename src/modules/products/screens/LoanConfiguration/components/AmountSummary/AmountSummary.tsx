import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { styles } from '../../LoanConfiguration.styles';

interface AmountSummaryProps {
  amount: string;
  onEditAmount: () => void;
}

export const AmountSummary: React.FC<AmountSummaryProps> = ({
  amount,
  onEditAmount,
}) => {
  return (
    <View style={styles.amountSummary}>
      <View style={styles.amountRow}>
        <Text style={styles.amountValue}>{amount ? `R$ ${amount}` : 'R$ 0,00'}</Text>
        <TouchableOpacity onPress={onEditAmount} style={styles.editButton}>
          <Ionicons name="pencil" size={16} color="#005CA9" />
        </TouchableOpacity>
      </View>
    </View>
  );
};
