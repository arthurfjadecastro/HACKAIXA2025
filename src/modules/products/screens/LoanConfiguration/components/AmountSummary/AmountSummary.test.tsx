import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { AmountSummary } from './AmountSummary';

// Mock do Ionicons
jest.mock('@expo/vector-icons', () => ({
  Ionicons: ({ name, size, color, ...props }: any) => {
    const { View } = require('react-native');
    return <View testID={`icon-${name}`} {...props} />;
  },
}));

// Mock dos styles
jest.mock('../../LoanConfiguration.styles', () => ({
  styles: {
    amountSummary: { padding: 10 },
    amountRow: { flexDirection: 'row' },
    amountValue: { fontSize: 16 },
    editButton: { padding: 5 },
  },
}));

describe('AmountSummary', () => {
  const defaultProps = {
    amount: '50000',
    onEditAmount: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly with amount', () => {
    const { getByText, getByTestId } = render(<AmountSummary {...defaultProps} />);

    expect(getByText('R$ 50000')).toBeTruthy();
    expect(getByTestId('icon-pencil')).toBeTruthy();
  });

  it('should render default amount when amount is empty', () => {
    const { getByText } = render(<AmountSummary {...defaultProps} amount="" />);

    expect(getByText('R$ 0,00')).toBeTruthy();
  });

  it('should call onEditAmount when edit button is pressed', () => {
    const { getByTestId } = render(<AmountSummary {...defaultProps} />);

    const editButton = getByTestId('icon-pencil').parent;
    fireEvent.press(editButton);

    expect(defaultProps.onEditAmount).toHaveBeenCalledTimes(1);
  });

  it('should display formatted amount with R$ prefix', () => {
    const { getByText } = render(<AmountSummary {...defaultProps} amount="100000" />);

    expect(getByText('R$ 100000')).toBeTruthy();
  });

  it('should render edit icon', () => {
    const { getByTestId } = render(<AmountSummary {...defaultProps} />);

    expect(getByTestId('icon-pencil')).toBeTruthy();
  });

  it('should handle zero amount', () => {
    const { getByText } = render(<AmountSummary {...defaultProps} amount="0" />);

    expect(getByText('R$ 0')).toBeTruthy();
  });

  it('should handle null amount', () => {
    const { getByText } = render(<AmountSummary {...defaultProps} amount="" />);

    expect(getByText('R$ 0,00')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const component = render(<AmountSummary {...defaultProps} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with large amounts', () => {
    const { getByText } = render(<AmountSummary {...defaultProps} amount="999999" />);

    expect(getByText('R$ 999999')).toBeTruthy();
  });

  it('should have correct structure', () => {
    const { getByText, getByTestId } = render(<AmountSummary {...defaultProps} />);

    expect(getByText('R$ 50000')).toBeTruthy();
    expect(getByTestId('icon-pencil')).toBeTruthy();
  });
});
