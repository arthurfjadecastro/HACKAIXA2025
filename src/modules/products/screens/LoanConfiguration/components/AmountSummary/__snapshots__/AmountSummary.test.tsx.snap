// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountSummary should match snapshot 1`] = `
<View
  style={
    {
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
        }
      }
    >
      R$ 50000
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        testID="icon-pencil"
      />
    </View>
  </View>
</View>
`;
